pysph/sph/wc/gtvf.py

```diff
@@ -0,0 +1,655 @@
+
+        fluids: list
+            List of names of fluid particle arrays.
+        solids: list
+            List of names of solid particle arrays (or boundaries).
+        dim: int
+            Dimensionality of the problem.
+        rho0: float
+            Reference density.
+        c0: float
+            Reference speed of sound.
+        nu: float
+            Real viscosity of the fluid.
+        h0: float
+            Reference smoothing length.
+        p0: float
+            reference pressure for the equation of state.
+        pref: float
+            reference pressure for rate of change of transport velocity.
+        gx: float
+            Body force acceleration components in x direction.
+        gy: float
+            Body force acceleration components in y direction.
+        gz: float
+            Body force acceleration components in z direction.
+        b: float
+            constant for the equation of state.
+        """
+
+        self.fluids = fluids
+        self.solids = solids
+        self.dim = dim
+        self.rho0 = rho0
+        self.c0 = c0
+        self.nu = nu
+        self.h0 = h0
+        self.p0 = p0
+        self.pref = pref
+        self.gx = gx
+        self.gy = gy
+        self.gz = gz
+        self.b = b
+        self.solver = None
+
+    def configure_solver(self, kernel=None, integrator_cls=None,
+                         extra_steppers=None, **kw):
+        """Configure the solver to be generated.
+
+        Parameters
+        ----------
+
+        kernel : Kernel instance.
+            Kernel to use, if none is passed a default one is used.
+        integrator_cls : pysph.sph.integrator.Integrator
+            Integrator class to use, use sensible default if none is
+            passed.
+        extra_steppers : dict
+            Additional integration stepper instances as a dict.
+        **kw : extra arguments
+            Any additional keyword args are passed to the solver instance.
+        """
+        from pysph.base.kernels import WendlandQuintic
+        if kernel is None:
+            kernel = WendlandQuintic(dim=self.dim)
+        steppers = {}
+        if extra_steppers is not None:
+            steppers.update(extra_steppers)
+
+        step_cls = GTVFStep
+        for fluid in self.fluids:
+            if fluid not in steppers:
+                steppers[fluid] = step_cls()
+
+        if integrator_cls is not None:
+            cls = integrator_cls
+            print("Warning: GTVF Integrator is not being used.")
+        else:
+            cls = GTVFIntegrator
+        integrator = cls(**steppers)
+
+        from pysph.solver.solver import Solver
+        self.solver = Solver(
+            dim=self.dim, integrator=integrator, kernel=kernel,
+            output_at_times=[0, 0.2, 0.4, 0.8], **kw
+        )
+
+    def get_equations(self):
+        from pysph.sph.wc.transport_velocity import StateEquation
+        sources = self.fluids
+
+        eq1, stage1 = [], []
```

Comment:
So this also does not incorporate solids so we should either not allow users to pass them in the constructor or add support for it.

Reply:
removed it from the scheme for now.